tests/test_solver_based_model.py

```diff
@@ -958,6 +958,21 @@ def test_essential_reactions(self):
                 self.model.reactions.Biomass_Ecoli_core_N_LPAREN_w_FSLASH_GAM_RPAREN__Nmet2.lower_bound = 999999.
                 self.model.essential_reactions()
 
+        def test_essential_metabolites(self):
+            model = self.model.copy()
+            essential_metabolites_unbalanced = [m.id for m in model.essential_metabolites(force_steady_state=False)]
+            essential_metabolites_balanced = [m.id for m in model.essential_metabolites(force_steady_state=True)]
+            self.assertTrue(sorted(essential_metabolites_unbalanced) == sorted(ESSENTIAL_METABOLITES))
+            self.assertTrue(sorted(essential_metabolites_balanced) == sorted(ESSENTIAL_METABOLITES))
+
+            with self.assertRaises(cameo.exceptions.SolveError):
+                self.model.reactions.Biomass_Ecoli_core_N_LPAREN_w_FSLASH_GAM_RPAREN__Nmet2.lower_bound = 999999.
+                self.model.essential_metabolites(force_steady_state=False)
+
+            with self.assertRaises(cameo.exceptions.SolveError):
```

Comment:
@KristianJensen 
The doesn't hold for CPLEX interface.

1) ERROR: test_knock_out (test_solver_based_model.TestMetaboliteCPLEX)

   Traceback (most recent call last):
    tests/test_solver_based_model.py line 1078 in test_knock_out
      metabolite_a.knock_out(time_machine=tm)
    cameo/core/metabolite.py line 110 in knock_out
      self._relax_mass_balance_constrain(time_machine)
    cameo/core/metabolite.py line 60 in _relax_mass_balance_constrain
      undo=partial(setattr, self.constraint, "lb", self.constraint.lb))
    cameo/util.py line 309 in __call__
      output = do()
    /Users/joao/Documents/repos/optlang/optlang/cplex_interface.py line 264 in lb
      if self.ub is not None and value > self.ub:
   TypeError: unorderable types: NoneType() > int()